include/fastscapelib/flow/flow_graph_impl.hpp

```diff
@@ -109,7 +113,12 @@ namespace fastscapelib
                 m_donors = xt::ones<size_type>(donors_shape) * -1;
                 m_donors_count = xt::zeros<size_type>({ grid.size() });
 
+                m_storage_indices = xt::arange<size_type>(0, grid.size(), 1);
+                m_random_levels = indices_type({ 0, size() });
```

Comment:
`random_levels` is not really good naming.. I don't know how to best describe random access, or any order

Totally agree about memory consumption.

> we could instead (re)use one of the other options (not sure, though, as this may clearly impact performance, i.e., computing BFS or DFS cache, CPU caching & parallelization).

Yeah I would say to stick with data layout but in the same time we already jump in buffers when indexing neighbors data..